kuri_joystick_teleop/src/kuri_joystick_teleop.cpp

```diff
@@ -0,0 +1,392 @@
+#include "kuri_joystick_teleop.h"
+
+KuriJoystickTeleop::KuriJoystickTeleop() :
+  head_action_srv("head_controller/follow_joint_trajectory"),
+  eye_action_srv("eyelids_controller/follow_joint_trajectory") {
+  linear_max = 0.4; // meter / sec
+  angular_max = M_PI / 3; // rad / sec
+  head_tilt_speed = 2.0; // position units / sec (based on controlLoop_hz)
+  head_tilt_max = 0.3; // position units
+  head_tilt_min = -0.8; // position units
+  head_pan_speed = 2.0; // position units per control / sec (based on controlLoop_hz)
+  head_pan_max = 0.75; // position units
+  head_pan_min = -0.75; // position units
+  head_eye_movement_duration = 0.15; // secs
+  eye_closed_position = 0.41; // position units
+  eye_open_position = 0.0; // position units
+  controlLoop_hz = 100.0; // Hz
+
+  base_cmd_pub = nh.advertise<geometry_msgs::Twist>("mobile_base/commands/velocity", 1000); //"cmd_vel" may also work
+  joystick_sub = nh.subscribe("joy", 1000, &KuriJoystickTeleop::joystickCallback, this);
+  nav_vel_sub = nh.subscribe("kuri_navigation_command_velocity", 1, &KuriJoystickTeleop::navVelCallback, this);
+
+  head_controller_state_sub = nh.subscribe("/head_controller/state", 1000, &KuriJoystickTeleop::headControllerStateCallback, this);
```

Comment:
Not as sure about this, but also probably don't want stale messages?

Reply:
Yup, don't want stale messages. Done.